### PR TITLE
chore(cli): display recommended actions on "copilot deploy"

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/svc_deploy.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func (o *deploySvcOpts) generateWorkerServiceRecommendedActions() {
 	retrieveEnvVarCode := "const eventsQueueURI = process.env.COPILOT_QUEUE_URI"
 	actionRetrieveEnvVar := fmt.Sprintf(
 		`Update %s's code to leverage the injected environment variable "COPILOT_QUEUE_URI".
-In JavaScript you can write %s.`,
+  In JavaScript you can write %s.`,
```
Comment (Lou1415926) [inline internal/pkg/cli/svc_deploy.go:222]: Do you think we want to move the call to `generateWorkerServiceRecommendedActions` inside `RecommandedActions()`?

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/cli/svc_deploy.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy.go:222]: Yup, I'll submit a separate refactor PR on monday. 

I think recommended actions should be something like this:
```go
type reachable interface {
   Port() int
}
type publisher interface {
    Topics() []*manifest.Topics
}
type subscriber interface {
    Subscriptions() []*manifest.Subscription
}

var actions []string
if _, ok := mft.(reachable); ok {
   actions = append(actions, o.generateEndpointRecs()...)
}
if _, ok := mft.(publisher); ok {
   actions = append(actions, o.generatePublishRecs()...)
}
if _, ok := mft.(subscriber); ok {
   actions = append(actions, o.generateSubscribeRecs()...)
}
```

This way the moment a new manifest allows the property of being reachable, or sns publishing then recommended actions will get the suggestion for free.

Hope that makes sense!

Comment (Lou1415926) [reply internal/pkg/cli/svc_deploy.go:222]: This is cool!! Looking forward to it!

**Review by iamhopaul123 — APPROVED** [review]
:shipit:

File: internal/pkg/cli/svc_deploy.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy.go:222]: Here is the commit with the changes: https://github.com/efekarakus/copilot-cli/commit/689745434b06ccdc859ec8e78ce1127ee98d175b (PR#2787)

**Pull request merged**